Building the native-image-agent-library macro fails with JDK 11.0.11+2

**Describe the issue**
When trying to build graal and then running the `native-image, --macro:native-image-agent-library` macro fails with:

```
Error: Could not find target method: private java.lang.invoke.MethodHandle com.oracle.svm.methodhandles.Target_java_lang_invoke_MethodHandles_Lookup.maybeBindCaller(com.oracle.svm.methodhandles.Target_java_lang_invoke_MemberName,java.lang.invoke.MethodHandle,java.lang.Class) throws java.lang.IllegalAccessException
```

**Steps to reproduce the issue**
1. build graal master with JDK 11.0.11+2 (e.g. via mandrel-packaging)
2. run `native-image, --macro:native-image-agent-library` after
3. Failure

**Describe GraalVM and your environment:**
 - GraalVM version: ** Graal master, revision 8bbae70e3ad4cc136d2042390b18e829230b7ded**
 - JDK major version:  11.0.11+2
 - OS: Linux
 - Architecture: **AMD64**


**More details**
This is caused by https://bugs.openjdk.java.net/browse/JDK-8235351 which is new in OpenJDK 11.0.11.

See also: https://github.com/graalvm/mandrel/issues/214 where we originally discovered this.


```
DEBUG [build] Launcher line AFTER: exec "${location}/../../../bin/java" --add-exports=java.base/jdk.internal.module=ALL-UNNAMED -XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCI -Dorg.graalvm.version="JDK-8247922-8616-gf38cc1648c2" -Dorg.graalvm.config="(Mandrel Distribution)" --upgrade-module-path ${location}/../../jvmci/graal.jar --add-modules "org.graalvm.truffle,org.graalvm.sdk" --module-path ${location}/../../truffle/truffle-api.jar:${location}/../../jvmci/graal-sdk.jar  "${jvm_args[@]}" -cp "${cp}" 'com.oracle.svm.driver.NativeImage$JDK9Plus' -J--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code=jdk.internal.vm.compiler  "${launcher_args[@]}"
DEBUG [OperatingSystem] Execute [/disk/graal/upstream-sources/graal/mandrel-build/lib/svm/bin/native-image, --macro:native-image-agent-library] in /disk/graal/upstream-sources/mandrel-packaging
Error: Could not find target method: private java.lang.invoke.MethodHandle com.oracle.svm.methodhandles.Target_java_lang_invoke_MethodHandles_Lookup.maybeBindCaller(com.oracle.svm.methodhandles.Target_java_lang_invoke_MemberName,java.lang.invoke.MethodHandle,java.lang.Class) throws java.lang.IllegalAccessException
Error: Use -H:+ReportExceptionStackTraces to print stacktrace of underlying exception
Error: Image build request failed with exit status 1
Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: Failed, exit code: 1
	at OperatingSystem.exec(build.java:1651)
	at build.buildAgents(build.java:176)
	at build.main(build.java:159)
Caused by: java.lang.RuntimeException: Failed, exit code: 1
	at OperatingSystem.exec(build.java:1642)
	at build.buildAgents(build.java:176)
	at build.main(build.java:159)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at jdk.compiler/com.sun.tools.javac.launcher.Main.execute(Main.java:404)
	at jdk.compiler/com.sun.tools.javac.launcher.Main.run(Main.java:179)
	at jdk.compiler/com.sun.tools.javac.launcher.Main.main(Main.java:119)
```

--- a/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/JDK11_0_10OrEarlier.java
+++ b/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/JDK11_0_10OrEarlier.java
@@ -1,0 +1,38 @@
+/*
+ * Copyright (c) 2021, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.core.jdk;
+
+import java.util.function.BooleanSupplier;
+
+import org.graalvm.compiler.serviceprovider.JavaVersionUtil;
+import org.graalvm.compiler.serviceprovider.GraalServices;
+
+public class JDK11_0_10OrEarlier implements BooleanSupplier {
+    @Override
+    public boolean getAsBoolean() {
+        return JavaVersionUtil.JAVA_SPEC < 11 ||
+                        (JavaVersionUtil.JAVA_SPEC == 11 && GraalServices.getJavaUpdateVersion() <= 10);
+    }
+}

--- a/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/JDK11_0_11OrLater.java
+++ b/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/JDK11_0_11OrLater.java
@@ -1,0 +1,38 @@
+/*
+ * Copyright (c) 2021, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.core.jdk;
+
+import java.util.function.BooleanSupplier;
+
+import org.graalvm.compiler.serviceprovider.JavaVersionUtil;
+import org.graalvm.compiler.serviceprovider.GraalServices;
+
+public class JDK11_0_11OrLater implements BooleanSupplier {
+    @Override
+    public boolean getAsBoolean() {
+        return JavaVersionUtil.JAVA_SPEC > 11 ||
+                        (JavaVersionUtil.JAVA_SPEC == 11 && GraalServices.getJavaUpdateVersion() > 10);
+    }
+}

--- a/substratevm/src/com.oracle.svm.methodhandles/src/com/oracle/svm/methodhandles/Target_java_lang_invoke_MethodHandles_Lookup.java
+++ b/substratevm/src/com.oracle.svm.methodhandles/src/com/oracle/svm/methodhandles/Target_java_lang_invoke_MethodHandles_Lookup.java
@@ -34,8 +34,8 @@ import com.oracle.svm.core.annotate.RecomputeFieldValue;
 import com.oracle.svm.core.annotate.Substitute;
 import com.oracle.svm.core.annotate.TargetClass;
 import com.oracle.svm.core.annotate.TargetElement;
-import com.oracle.svm.core.jdk.JDK11OrEarlier;
-import com.oracle.svm.core.jdk.JDK15OrLater;
+import com.oracle.svm.core.jdk.JDK11_0_10OrEarlier;
+import com.oracle.svm.core.jdk.JDK11_0_11OrLater;
 import com.oracle.svm.core.jdk.JDK16OrLater;
 
 @TargetClass(value = MethodHandles.class, innerClass = "Lookup", onlyWith = MethodHandlesSupported.class)
@@ -48,7 +48,7 @@ final class Target_java_lang_invoke_MethodHandles_Lookup {
 
     @SuppressWarnings({"static-method", "unused"})
     @Substitute
-    @TargetElement(onlyWith = JDK11OrEarlier.class)
+    @TargetElement(onlyWith = JDK11_0_10OrEarlier.class)
     private MethodHandle maybeBindCaller(Target_java_lang_invoke_MemberName method, MethodHandle mh,
                     Class<?> boundCallerClass)
                     throws IllegalAccessException {
@@ -58,7 +58,7 @@ final class Target_java_lang_invoke_MethodHandles_Lookup {
 
     @SuppressWarnings({"static-method", "unused"})
     @Substitute
-    @TargetElement(onlyWith = JDK15OrLater.class)
+    @TargetElement(onlyWith = JDK11_0_11OrLater.class)
     private MethodHandle maybeBindCaller(Target_java_lang_invoke_MemberName method, MethodHandle mh,
                     Target_java_lang_invoke_MethodHandles_Lookup boundCaller)
                     throws IllegalAccessException {

